aiokafka/conn.py

```diff
@@ -498,20 +502,23 @@ def _create_reader_task(self):
             functools.partial(self._on_read_task_error, self_ref))
         return read_task
 
-    @classmethod
-    async def _read(cls, self_ref):
+    @staticmethod
+    async def _read(self_ref):
         # XXX: I know that it become a bit more ugly once cyclic references
         # were removed, but it's needed to allow connections to properly
         # release resources if leaked.
         # NOTE: all errors will be handled by done callback
+        self = self_ref()
```

Comment:
This will create a dangling reference on the task for all time. It defeats the purpose of having a weekref

Reply:
You will have to remove the reference before entering the for look